packages/client/hmi-client/src/components/dataset/tera-dataset-jupyter-panel.vue

```diff
@@ -234,11 +245,49 @@ jupyterSession.kernelChanged.connect((_context, kernelInfo) => {
 	if (kernel?.name === 'beaker_kernel') {
 		setKernelContext(kernel as IKernelConnection, {
 			context: 'dataset',
+			language: selectedLanguage.value,
 			context_info: contextInfo
 		});
 	}
 });
 
+const onChangeLanguage = (val) => {
+	confirm.require({
+		message:
+			'Are you sure you want to change the language? Changing the language will reset the kernel to its starting state.',
+		header: 'Change language',
+		icon: 'pi pi-exclamation-triangle',
+		accept: () => {
+			const session = jupyterSession.session;
+			const kernel = session?.kernel as IKernelConnection;
+
+			const contextInfo: any = {};
+			props.assets.forEach((asset, i) => {
+				const key = `d${i + 1}`;
+				contextInfo[key] = {
+					id: asset.id,
+					asset_type: toAssetType(asset.type)
+				};
+			});
+			const messageBody = {
+				session: session?.name || '',
+				channel: 'shell',
+				content: {
+					context: 'dataset',
+					language: val.value,
+					context_info: contextInfo
+				},
+				msgType: 'context_setup_request',
+				msgId: createMessageId('context_setup_request')
+			};
+			const message: JupyterMessage = createMessage(messageBody);
+			kernel.sendJupyterMessage(message);
+
+			emit('update-language', val.value);
```

Comment:
Does ordering matter? I don't know if you have the guarantee that `kernel.sendJupyterMessage(message);` is finished at this stage.

Reply:
You're right, I'm making a change to listen to the `context_setup_response` to get the language, that should give use the language change

Reply:
@mwdchang it _should_ be good now with the latest commit